src/Compute/Compute/VirtualMachine/Config/AddAzureVMDataDiskCommand.cs

```diff
@@ -144,6 +145,13 @@ public class AddAzureVMDataDiskCommand : ComputeClientBaseCmdlet
         [PSArgumentCompleter("Detach", "Delete")]
         public string DeleteOption { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "ARM ID of snapshot or disk restore point from which to create a disk.")]
+        [ValidateNotNullOrEmpty]
+        public string SourceResourceId { get; set; }
```

Comment:
Should this be a string or a ResourceId type? Or some other guid.

Reply:
Should be a string, I was thinking of then using that resource id in a ResourceIdentifier object (like in Get-AzVmss) in ps to help parse it, but that's not needed here.